infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -130,38 +149,40 @@ private long getStartingSegment() {
      * Create stream log directory if not exists
      */
     private void initStreamLogDirectory() {
-        if (logDir.toFile().exists()) {
+
+        try {
+            if (!logDir.toFile().exists()) {
+                Files.createDirectories(logDir);
+            }
+
             String corfuDir = logDir.getParent().toString();
-            // If FileSystem is mounted as read-only, Corfu server cannot function.
-            try {
-                FileStore fs = Files.getFileStore(Paths.get(corfuDir));
-                if (fs.isReadOnly()) {
-                    throw new UnrecoverableCorfuError("Cannot start Corfu on a read-only filesystem:"
-                            +corfuDir);
-                }
-            } catch (IOException e) {
-                throw new UnrecoverableCorfuError("Unable to retrieve Corfu Filesystem permissions"+
-                        corfuDir);
+            FileStore fs = Files.getFileStore(Paths.get(corfuDir));
+
+            if (fs.isReadOnly()) {
+                throw new IllegalArgumentException("Cannot start Corfu on a read-only filesystem:"
```

Comment:
There is no argument here. Is RuntimeException better?

Reply:
I see your point. Changed to `LogUnitException`